### PR TITLE
WT-9372 Add a debug config option to turn cursor repositioning on-off

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

**Comment by sulabhM** [comment]
@Siddhartha8899 , the config option is ready. By default it is off. You can use it by adding to the wiredtiger connection string, `"debug_mode=[cursor_reposition=true]"`.

For instance, I can run a set of python tests with the above string passed as command line parameter:

```WIREDTIGER_CONFIG="debug_mode=[cursor_reposition=true]" WT_BUILDDIR=$(pwd) python3 ../test/suite/run.py -v 16 txn```

File: dist/api_data.py [hunk 1]
```diff
@@ -566,6 +566,12 @@ def __ge__(self, other):
             operation. This allows memory sanitizers to detect inappropriate references to
             memory owned by cursors.''',
             type='boolean'),
+        Config('cursor_reposition', 'false', r'''
+            if true, for the operations with snapshot isolation the cursor temporarily releases the
+            page it would otherwise keep pinned, then repositions back to the page for further
```
Comment (quchenhao) [inline dist/api_data.py:571]: It only needs to release pages that require force eviction right?

File: src/config/config_def.c [hunk 2]
```diff
@@ -62,7 +62,9 @@ static const WT_CONFIG_CHECK confchk_WT_CONNECTION_reconfigure_compatibility_sub
 static const WT_CONFIG_CHECK confchk_wiredtiger_open_debug_mode_subconfigs[] = {
   {"checkpoint_retention", "int", NULL, "min=0,max=1024", NULL, 0},
   {"corruption_abort", "boolean", NULL, NULL, NULL, 0},
-  {"cursor_copy", "boolean", NULL, NULL, NULL, 0}, {"eviction", "boolean", NULL, NULL, NULL, 0},
+  {"cursor_copy", "boolean", NULL, NULL, NULL, 0},
+  {"cursor_reposition", "boolean", NULL, NULL, NULL, 0},
+  {"eviction", "boolean", NULL, NULL, NULL, 0},
```
Comment (quchenhao) [inline src/config/config_def.c:67]: What's this change for eviction?

Comment (Siddhartha8899) [reply src/config/config_def.c:67]: This is not a new change. It existed before as well. Just on one line. [here](https://github.com/wiredtiger/wiredtiger/pull/7977/files#diff-ff528f30fd5f654ae4d06b4cf237b1f3e4e57be76a432dcbb00e91c13e646d52L65)

File: dist/api_data.py (hunk 1 above)
Comment (Siddhartha8899) [reply dist/api_data.py]: Updated in [a4aa307](https://github.com/wiredtiger/wiredtiger/pull/7977/commits/a4aa307e3659676a9ae22e4709e5bda62ff5927f). Thanks!

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Review by agorrod — APPROVED** [review]
LGTM

**Pull request merged**